sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala

```diff
@@ -292,7 +292,10 @@ object FileFormatWriter extends Logging {
     override def execute(iter: Iterator[InternalRow]): Set[String] = {
       var fileCounter = 0
       var recordsInFile: Long = 0L
-      newOutputWriter(fileCounter)
+      // Skip the empty partition to avoid creating a mass of 'empty' files.
+      if (iter.hasNext) {
+        newOutputWriter(fileCounter)
```

Comment:
Reading empty data should be fine too. I should preserve the schema. I am pretty sure that we want this case because mine was reverted due to the case above.